tools/depmat.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python3
+
+import sys
+from getopt import gnu_getopt, GetoptError
+from re import compile as Regex
+from collections import namedtuple
+
+from libgcopdd import Event, UserError, args_to_events, n_args_to_n_events, open_dump_entry, create_temp
+import phasestack
+
+INTEGER_PATTERN = Regex(r'\d+')
+DepMat = namedtuple('DepMat', 'size, mapping, data')
+
+####
+
+def load(event):
+	"""Loads `.depmat` dump entry of given `event`.
+	
+	The `event` contains the dump name and event hash,
+	but not the `.depmat` suffix. Returns a DepMat object.
+	"""
+	with open_dump_entry(event + '.depmat') as file:
+		mapping = []
+		for line in file:
+			if line.isspace():
+				break
+			mapping.append(int(line))
+		size = len(mapping)
+		data = list(map(int, INTEGER_PATTERN.findall(file.read())))
+	assert len(data) == 3*size*size
+	return DepMat(size, mapping, data)
+
+def save(depmat, event):
+	"""Saves `.depmat` dump entry to given `event`.
+	
+	The `event` contains the dump name and event hash,
+	but not the `.depmat` suffix. `depmat` is a DepMat object.
+	"""
+	with open_dump_entry(event + '.depmat', 'wt') as file:
+		rowsize = 3 * depmat.size
+		for value in depmat.mapping or range(depmat.size):
+			file.write(str(value) + '\n')
+		for idx, value in enumerate(depmat.data):
+			if idx % 3:
+				file.write(':' + str(value))
+			elif idx % rowsize:
+				file.write(' ' + str(value))
+			else:
+				file.write('\n' + str(value))
+		file.write('\n')
+
+####
+
+def add_to(matrix, addend):
```

Comment:
This looks complicated. On a first sight, I don't know what a matrix is (how is it represented), so I have a hard time trying to understand this.

Reply:
Explanation of matrix data type would probably help.